packages/runtime/src/runtimes/react/react-runtime.ts

```diff
@@ -50,9 +85,30 @@ export class ReactRuntime extends RuntimeCore {
     }
   }
 
-  constructor({ breakpoints }: { breakpoints?: BreakpointsInput } = {}) {
+  constructor({
+    breakpoints,
+    components,
+  }: {
+    breakpoints?: BreakpointsInput
+    components?: Record<string, ComponentRegistration<any>>
+  } = {}) {
     super({ breakpoints })
 
     registerBuiltinComponents(this)
```

Comment:
What if, when you pass `components`, we don't register the built-in components by default, and you have to include the default components manually?

We need a way to do this for the [Remove built-in components from runtime](https://linear.app/makeswift/project/remove-built-in-components-from-runtime-fe5104cd48a3) project

Reply:
Would we export the builtin components so you can do something like `[…myComponents, …builtinComponents]`?

Reply:
Yes, we need to export the builtin components, or at least the registration.

We might want to make it explicit though, so users can easily remove components that they don't need, like how we want to exclude Form on Catalyst.

Reply:
Yeah, perhaps we can have an option, `disableAutomaticBuiltinsRegistration` on the runtime constructor